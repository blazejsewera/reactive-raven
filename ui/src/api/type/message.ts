export type ApiMessage = {
  username: string
  timestamp: string
  title: string
  subtitle?: string
  body?: string
  id?: string
}

// example:
// { "username": "Test", "timestamp": "a", "title": "Test title", "subtitle": "Test subtitle", "body": "Test body.", "id": "1234" }
