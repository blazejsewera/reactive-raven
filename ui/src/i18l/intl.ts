/**
 * Intl is an interface for an internationalization (i18l) tool
 * that can provide translations for page.
 */
export interface Intl {
  getTranslation: (text: string) => string
}

export const intl: Intl = { getTranslation: text => text }
