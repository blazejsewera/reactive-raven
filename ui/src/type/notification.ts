import { NotificationCardHandlers } from './handler'

export type Notification = {
  appName: string
  timestamp: string
  appImgUri?: string
  title: string
  subtitle?: string
  body?: string
  id?: string
  handlers: NotificationCardHandlers
}

export type NotificationWithHandlers = {
  notification: Notification
  handlers: NotificationCardHandlers
}
