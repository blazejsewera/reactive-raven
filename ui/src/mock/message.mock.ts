import { Message } from '../type/message'
import { fiveSentenceLoremIpsum } from './asset/text/lipsum'
import githubIcon from './asset/icon/github-icon.svg'

const body = `This is a simple test message.

It spans over multiple lines.`

export const full: Message = {
  username: 'Primary',
  title: 'Test title',
  subtitle: '— primary',
  body,
  timestamp: '2 hours ago',
  id: '0',
}

export const fullWithImage: Message = {
  ...full,
  id: '1',
  userImgUri: githubIcon,
}

export const fullWithLoremIpsum: Message = {
  ...full,
  id: '2',
  body: fiveSentenceLoremIpsum,
}

export const partial: Message = {
  username: full.username,
  title: full.title,
  body: full.body,
  timestamp: full.timestamp,
  id: '3',
}

export const minimal: Message = {
  username: full.username,
  title: full.title,
  timestamp: full.timestamp,
  id: '4',
}

export const anotherUser: Message = {
  username: 'Other',
  title: 'Other title',
  subtitle: '— other',
  body,
  timestamp: '1 hour ago',
  id: '5',
}
