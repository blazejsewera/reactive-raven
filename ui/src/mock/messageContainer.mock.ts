import type { MessageContainerHandlers } from '../type/handler'

export const handlers: MessageContainerHandlers = {
  onClear: () => {},
}
