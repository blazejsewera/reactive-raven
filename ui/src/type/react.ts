import type { ReactElement, ReactNode } from "react";

type EmptyObject = Record<string, never>;

/**
 * Default children type in React. Every renderable type is included.
 */
export type DefaultChildren = ReactNode;

/**
 * No props, a.k.a. () => </> component
 */
export type EmptyProps = EmptyObject;

/**
 * No React Context
 */
export type EmptyContext = EmptyObject;

/**
 * Functional Component (strongly typed)
 */
export interface FC<P = EmptyProps, Ctx = EmptyContext> {
  (props: P, context?: Ctx): ReactElement | null;
  displayName?: string;
}
