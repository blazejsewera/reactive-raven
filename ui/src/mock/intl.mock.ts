import { Intl } from '../i18l/intl'

export const intlMock: Intl = { getTranslation: text => text }
