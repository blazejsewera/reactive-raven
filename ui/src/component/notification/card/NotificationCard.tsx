import * as React from 'react'
import type { FC } from '../../../type/react'
import type { Notification } from '../../../type/notification'
import type { Intl } from '../../../i18l/intl'
import { cx } from '../../../util/cx'
import { Header } from './section/Header'
import { Body } from './section/Body'
import { Footer } from './section/Footer'
import { Indicator } from './sprite/icon/Indicator'

export interface NotificationCardProps {
  notification: Notification
  intl: Intl
}

export const NotificationCard: FC<NotificationCardProps> = ({ notification, intl }) => {
  const { appName, appImgUri, title, subtitle, body, timestamp, handlers } = notification

  return (
    <div className={cx('rounded-3xl', 'bg-white', 'dark:bg-gray-800', 'w-80', 'p-5', 'shadow-lg', 'relative')}>
      <Header {...{ appName, appImgUri, title, subtitle }} />
      <Indicator />
      <Body intl={intl}>{body ?? ''}</Body>
      <Footer {...{ appName, timestamp, handlers }} />
    </div>
  )
}
