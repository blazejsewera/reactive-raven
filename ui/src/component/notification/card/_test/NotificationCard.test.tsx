import * as React from 'react'
import { create as render } from 'react-test-renderer'
import type { Notification } from '../../../../type/notification'
import { NotificationCard } from '../NotificationCard'
import { intlMock } from '../../../../mock/intl.mock'
import { full, fullWithImage, fullWithLoremIpsum, minimal, partial } from '../../../../mock/notification.mock'
import type { NotificationCardHandlers } from '../../../../type/handler'

describe('NotificationCard component', () => {
  // given
  const intl = intlMock
  const handlers: NotificationCardHandlers = {
    onArchive: jest.fn(),
    onCheck: jest.fn(),
    onSettings: jest.fn(),
  }

  const testNotificationCard = (notification: Notification) => {
    const tree = render(<NotificationCard {...{ notification, intl, handlers }} />).toJSON()
    expect(tree).toMatchSnapshot()
  }

  it('renders correctly with full notification', () => {
    testNotificationCard(full)
  })
  it('renders correctly with full with image notification', () => {
    testNotificationCard(fullWithImage)
  })
  it('renders correctly with full with very long text notification', () => {
    testNotificationCard(fullWithLoremIpsum)
  })
  it('renders correctly with partial notification', () => {
    testNotificationCard(partial)
  })
  it('renders correctly with minimal notification', () => {
    testNotificationCard(minimal)
  })
})
