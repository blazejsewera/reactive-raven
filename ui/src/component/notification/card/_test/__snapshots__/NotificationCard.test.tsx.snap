// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCard component renders correctly with full notification 1`] = `
<div
  className="rounded-3xl bg-white dark:bg-gray-800 w-80 p-5 shadow-lg relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="#000"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800 dark:text-gray-300"
      >
        New Pull Request
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — notipie
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-2.5 h-2.5 bg-yellow-500 dark:bg-yellow-600 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800 dark:text-gray-300"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component renders correctly with full with image notification 1`] = `
<div
  className="rounded-3xl bg-white dark:bg-gray-800 w-80 p-5 shadow-lg relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default border dark:bg-gray-300 dark:border-gray-300"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800 dark:text-gray-300"
      >
        New Pull Request
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — notipie
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-2.5 h-2.5 bg-yellow-500 dark:bg-yellow-600 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800 dark:text-gray-300"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component renders correctly with full with very long text notification 1`] = `
<div
  className="rounded-3xl bg-white dark:bg-gray-800 w-80 p-5 shadow-lg relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="#000"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800 dark:text-gray-300"
      >
        New Pull Request
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — notipie
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-2.5 h-2.5 bg-yellow-500 dark:bg-yellow-600 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800 dark:text-gray-300"
  >
    <span>
      Explicabo ipsum iusto qui porro iste possimus ad voluptatem. Nam maxime voluptatem nostrum deleniti saepe. Officia dolor velit sunt velit placeat amet
      <span
        aria-label="show more"
        className="text-blue-500 dark:text-blue-400 cursor-pointer"
        onClick={[Function]}
      >
        [...]
      </span>
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component renders correctly with minimal notification 1`] = `
<div
  className="rounded-3xl bg-white dark:bg-gray-800 w-80 p-5 shadow-lg relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="#000"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800 dark:text-gray-300"
      >
        New Pull Request
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-2.5 h-2.5 bg-yellow-500 dark:bg-yellow-600 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800 dark:text-gray-300"
  >
    <span>
      
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`NotificationCard component renders correctly with partial notification 1`] = `
<div
  className="rounded-3xl bg-white dark:bg-gray-800 w-80 p-5 shadow-lg relative"
>
  <div
    className="flex"
  >
    <div
      className="h-12 w-12 rounded-full overflow-clip overflow-hidden inline-block cursor-default"
    >
      <svg
        viewBox="0 0 1 1"
      >
        <rect
          fill="#000"
          height={1}
          width={1}
        />
        <text
          dy="0.178"
          fill="#FFF"
          style={
            Object {
              "fontFamily": "Inter",
              "fontSize": 0.5,
              "fontWeight": "bold",
            }
          }
          textAnchor="middle"
          x={0.5}
          y={0.5}
        >
          Gi
        </text>
      </svg>
    </div>
    <div
      className="my-auto inline-block ml-5"
    >
      <p
        className="text-lg font-bold mb-0 leading-5 text-gray-800 dark:text-gray-300"
      >
        New Pull Request
      </p>
    </div>
  </div>
  <div
    className="rounded-full w-2.5 h-2.5 bg-yellow-500 dark:bg-yellow-600 absolute top-4 right-4"
  />
  <p
    className="whitespace-pre-line text-sm leading-4 my-3 text-gray-800 dark:text-gray-300"
  >
    <span>
      #12 add some new amazing functionality

Closes #10. Changes both in 'core' and 'ui'. Needs additional work with this and that.
    </span>
  </p>
  <div
    className="flex"
  >
    <div
      className="inline-block"
    >
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        — by 
        Github
      </p>
      <p
        className="text-xs font-bold text-gray-500 dark:text-gray-400"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="inline-grid grid-cols-3 gap-x-5 ml-auto my-auto mr-2"
    >
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="21 8 21 21 3 21 3 8"
          />
          <rect
            height="5"
            width="22"
            x="1"
            y="3"
          />
          <line
            x1="10"
            x2="14"
            y1="12"
            y2="12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="20 6 9 17 4 12"
          />
        </svg>
      </button>
      <button
        className="inline"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="stroke-current text-gray-500 dark:text-gray-400 w-5 h-5"
          fill="none"
          height={24}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          viewBox="0 0 24 24"
          width={24}
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
